src/lib.rs

```diff
@@ -58,18 +66,20 @@ impl<W, T> PistonWindow<W, T>
         use piston::event::Events;
         use piston::window::{ OpenGLWindow, Window };
 
-        let (mut device, mut factory) = gfx_device_gl::create(|s| window.borrow_mut().get_proc_address(s));
+        let mut device = gfx_device_gl::Device::new(|s| window.borrow_mut().get_proc_address(s));
+        let mut factory = device.spawn_factory();
```

Comment:
@mitchmindtree I believe it's going to work fine now after https://github.com/PistonDevelopers/gfx_graphics/pull/241 . Please let me know if you still have issues after it's merged.